pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -0,0 +1,356 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/go-openapi/strfmt"
+	apimodels "github.com/grafana/alerting-api/pkg/api"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/util"
+	amv2 "github.com/prometheus/alertmanager/api/v2/models"
+)
+
+type AlertmanagerSrv struct {
+	am    Alertmanager
+	store store.AlertingStore
+	log   log.Logger
+}
+
+func (srv AlertmanagerSrv) RouteCreateSilence(c *models.ReqContext, postableSilence apimodels.PostableSilence) response.Response {
+	silenceID, err := srv.am.CreateSilence(&postableSilence)
+	if err != nil {
+		if errors.Is(err, notifier.ErrSilenceNotFound) {
+			return response.Error(http.StatusNotFound, err.Error(), nil)
+		}
+
+		if errors.Is(err, notifier.ErrCreateSilenceBadPayload) {
+			return response.Error(http.StatusBadRequest, err.Error(), nil)
+		}
+
+		return response.Error(http.StatusInternalServerError, "failed to create silence", err)
+	}
+	return response.JSON(http.StatusAccepted, util.DynMap{"message": "silence created", "id": silenceID})
+}
+
+func (srv AlertmanagerSrv) RouteDeleteAlertingConfig(c *models.ReqContext) response.Response {
+	// not implemented
+	return response.Error(http.StatusNotImplemented, "", nil)
+}
+
+func (srv AlertmanagerSrv) RouteDeleteSilence(c *models.ReqContext) response.Response {
+	silenceID := c.Params(":SilenceId")
+	if err := srv.am.DeleteSilence(silenceID); err != nil {
+		if errors.Is(err, notifier.ErrSilenceNotFound) {
+			return response.Error(http.StatusNotFound, err.Error(), nil)
+		}
+		return response.Error(http.StatusInternalServerError, err.Error(), nil)
+	}
+	return response.JSON(http.StatusOK, util.DynMap{"message": "silence deleted"})
+}
+
+func (srv AlertmanagerSrv) RouteGetAlertingConfig(c *models.ReqContext) response.Response {
+	query := ngmodels.GetLatestAlertmanagerConfigurationQuery{}
+	err := srv.store.GetLatestAlertmanagerConfiguration(&query)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to get latest configuration", err)
+	}
+
+	cfg := apimodels.PostableUserConfig{}
+	err = yaml.Unmarshal([]byte(query.Result.AlertmanagerConfiguration), &cfg)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to unmarshal alertmanager configuration", err)
+	}
+
+	var apiReceiverName string
+	var receivers []*apimodels.GettableGrafanaReceiver
+	alertmanagerCfg := cfg.AlertmanagerConfig
+	if len(alertmanagerCfg.Receivers) > 0 {
+		apiReceiverName = alertmanagerCfg.Receivers[0].Name
```

Comment:
Not sure I follow this, why do we have to use the first receiver instead of looping through them?

Reply:
The structs are:

https://github.com/grafana/alerting-api/blob/3294563b51bb9898727952f7382f445ca526b653/pkg/api/alertmanager.go#L388-L391

https://github.com/grafana/alerting-api/blob/3294563b51bb9898727952f7382f445ca526b653/pkg/api/alertmanager.go#L467-L471

so this  field is the [Name](https://github.com/grafana/grafana/blob/5c036fa6ef7d0874d0313f8f8d460cb298132488/pkg/services/ngalert/api/test-data/post-user-config.json#L17) field of the `config.Receiver` embedded type as opposed to the [Name](https://github.com/grafana/grafana/blob/5c036fa6ef7d0874d0313f8f8d460cb298132488/pkg/services/ngalert/api/test-data/post-user-config.json#L21) field of each `PostableGrafanaReceiver`